src/hal/src/format.rs

```diff
@@ -255,11 +255,14 @@ surface_types! {
     R4_G4               {  !8, COLOR, (1, 1), color: 8 },
     R4_G4_B4_A4         { !16, COLOR, (1, 1), color: 12, alpha: 4 },
     B4_G4_R4_A4         { !16, COLOR, (1, 1), color: 12, alpha: 4 },
+    A4_B4_G4_R4         { !16, COLOR, (1, 1), color: 12, alpha: 4 },
```

Comment:
hold on, we don't want to introduce any new HAL formats. Those are 1:1 mapped to Vulkan for the sake of portability initiative. If something can't be mapped to Metal, that becomes a problem of the Metal backend, not HAL.